components/AlgoliaSearch.tsx

```diff
@@ -0,0 +1,321 @@
+function Hit({ hit, children }: IHitProps) {
+  return (
+    <Link
+      href={hit.url}
+      className={clsx({
+        'DocSearch-Hit--Result': hit.__is_result?.(),
+        'DocSearch-Hit--Parent': hit.__is_parent?.(),
+        'DocSearch-Hit--FirstChild': hit.__is_first?.(),
+        'DocSearch-Hit--LastChild': hit.__is_last?.(),
+        'DocSearch-Hit--Child': hit.__is_child?.()
+      })}
+    >
+      {children}
+    </Link>
+  );
+}
+
+/**
+ *
+ * @param {IAlgoliaModalProps} props - The props of the Algolia modal
+ * @description The Algolia modal used for searching the website
+ */
+function AlgoliaModal({ onClose, initialQuery, indexName }: AlgoliaModalProps) {
+  const router = useRouter();
+
+  return createPortal(<DocSearchModal
+    initialQuery={initialQuery}
+    initialScrollY={window.scrollY}
+    searchParameters={{
+      distinct: 1
+    }}
+    placeholder={indexName === DOCS_INDEX_NAME ? 'Search documentation' : 'Search resources'}
+    onClose={onClose}
+    indexName={indexName}
+    apiKey={API_KEY}
+    appId={APP_ID}
+    navigator={{
+      navigate({ itemUrl }) {
+        onClose();
+        router.push(itemUrl);
+      }
+    }}
+    hitComponent={Hit}
+    transformItems={transformItems}
+    getMissingResultsUrl={({ query }) => {
+      return `https://github.com/asyncapi/website/issues/new?title=Cannot%20search%20given%20query:%20${query}`;
+    }}
+  />,
+  document.body);
+}
+
+/**
+ * @param {KeyboardEvent} event - The keyboard event
+ * @returns {boolean} - Whether the content is being edited
+ * @description The function used to check if the content is being edited
+ */
+function isEditingContent(event: KeyboardEvent) {
+  const element = event.target;
+  const { tagName } = element as HTMLElement;
+
+  return (
+    (element as HTMLElement).isContentEditable ||
+    tagName === 'INPUT' ||
+    tagName === 'SELECT' ||
+    tagName === 'TEXTAREA'
+  );
+}
+
+/**
+ * @returns {Object} - The action key
+ * @description The function used to get the action key
+ */
+function getActionKey() {
+  if (typeof navigator !== 'undefined') {
+    if (/(Mac|iPhone|iPod|iPad)/i.test(navigator.userAgent || navigator.platform)) {
+      return {
+        shortKey: '⌘',
+        key: 'Command'
+      };
+    }
+
+    return {
+      shortKey: 'Ctrl',
+      key: 'Control'
+    };
+  }
+
+  return {
+    shortKey: 'Ctrl',
+    key: 'Control'
+  };
+}
+
+/**
+ *
+ * @param {IUseDocSearchKeyboardEvents} props - The props of the useDocSearchKeyboardEvents hook
+ * @description The hook used for the Algolia search keyboard events
+ */
+function useDocSearchKeyboardEvents({ isOpen, onOpen, onClose }: IUseDocSearchKeyboardEvents) {
+  useEffect(() => {
+    /**
+     * @param {KeyboardEvent} event - The keyboard event
+     * @returns {void}
+     * @description The function used to handle the keyboard event
+     */
+    function onKeyDown(event: KeyboardEvent): void {
+      if (
+        (event.keyCode === 27 && isOpen) ||
```

Comment:
Kindly remove the keyCode implementation as it is now deprecated. Reference - https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode.

Reply:
Sure